hawkular-inventory-bus/src/main/java/org/hawkular/inventory/bus/MessageSender.java

```diff
@@ -35,24 +36,25 @@
  * @since 0.0.1
  */
 final class MessageSender {
-    private final MessageProcessor messageProcessor;
-    private final ConnectionContextFactory connectionContextFactory;
     private final String topicName;
-    private ProducerConnectionContext producerConnectionContext;
+    private final TopicConnectionFactory topicConnectionFactory;
+    private final MessageProcessor messageProcessor;
 
+    public MessageSender(TopicConnectionFactory  topicConnectionFactory, String topicName) {
```

Comment:
are you always going to be sending to a Topic? If you are never going to use queues, this is OK, but if you would ever want to send to a queue, you'd need to pass in ConnectionFactory not one of [Topic,Queue]ConnectionFactory